smart-contract-verifier/smart-contract-verifier-server/src/services/solidity_verifier.rs

```diff
@@ -82,53 +97,102 @@ impl SolidityVerifierService {
     }
 }
 
+fn process_verify_result(
+    result: Result<SoliditySuccess, VerificationError>,
+    post_actions: Vec<VerifyPostAction>,
+) -> Result<VerifyResponse, Status> {
+    match result {
+        Ok(res) => {
+            // Process requested post actions
+            let mut post_actions_responses: PostActionResponses = Default::default();
+            for action in post_actions {
+                match action {
+                    VerifyPostAction::LookupMethods => {
+                        let methods = find_methods_from_compiler_output(
+                            &res.contract_name,
+                            &res.compiler_output,
+                        );
+                        match methods {
+                            Ok(methods) => {
+                                let response = LookupMethodsResponseWrapper::from(methods);
+                                post_actions_responses.lookup_methods = Some(response.into());
+                            }
+                            Err(err) => {
+                                tracing::error!("lookup-methods error: {err:#?}");
+                            }
+                        }
+                    }
+                }
+            }
```

Comment:
maybe move to `fn process_post_actions(res, actions) -> PostActionResponses` ?

Reply:
Fixed